c2_components/src/mfx_c2_decoder_component.cpp

```diff
@@ -222,6 +221,13 @@ MfxC2DecoderComponent::MfxC2DecoderComponent(const C2String name, const CreateCo
         .withConstValue(AllocSharedString<C2PortMediaTypeSetting::output>("video/raw"))
         .build());
 
+    addParameter(
+        DefineParam(m_outputUsage, C2_PARAMKEY_OUTPUT_STREAM_USAGE)
+        .withDefault(new C2StreamUsageTuning::output(0u, C2AndroidMemoryUsage::HW_CODEC_WRITE))
```

Comment:
This is defined for consumer usage. Could you please use HW_TEXTURE_READ  and HW_COMPOSER_READ  for the default usage here?

 enum consumer_t : uint64_t {
         RENDERSCRIPT_READ = GRALLOC_USAGE_RENDERSCRIPT,
         HW_TEXTURE_READ   = GRALLOC_USAGE_HW_TEXTURE,
         HW_COMPOSER_READ  = GRALLOC_USAGE_HW_COMPOSER,
         // gralloc does not define a video decoder read usage flag, so use encoder for
         // now
         HW_CODEC_READ     = GRALLOC_USAGE_HW_VIDEO_ENCODER,
         READ_PROTECTED    = GRALLOC_USAGE_PROTECTED,
     };

Reply:
Updated